cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -663,6 +672,95 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
     return cw.toByteArray();
   }
 
+  /**
+   * Rewrites the temporary file name used for checkpoints from checkpointDir/temp to
+   * checkpointDir/temp-(current_timestamp_milliseconds).
```

Comment:
You are using the checkpoint time instead of the current timestamp, right?

Reply:
Correct. I've updated the comment.